spec/components/radio_spec.rb

```diff
@@ -119,6 +119,20 @@ def component_name
     assert_select "legend h1", "What is your favourite skittle?"
   end
 
+  it "renders radio-group with custom heading size" do
+    render_component(
+      name: "favourite-skittle",
+      heading: "What is your favourite skittle?",
+      heading_size: "s",
```

Comment:
What happens if an invalid value is passed in? e.g. "q"?

Reply:
[Will default to 'm'](https://github.com/alphagov/govuk_publishing_components/pull/1107/files#diff-fe62b3a78b7005dd4f069f961652b72cR10) which is the hard-coded value at the moment.

Reply:
@alex-ju are you sure?

Reply:
Not with the pushed code, no. Will update it to use an associated hash, full names for heading sizes and handle the default properly.

Reply:
Just tested it and it adds to the class whatever you pass. You could try something like:

```
css_classes << "govuk-fieldset__legend--#{heading_size}" if ['s', 'm', 'l', 'xl'].include?(heading_size)
```

Reply:
Yep, that would be consistent with the approach we use for margins.

Reply:
Updated to default to 'm' when both no size is being passed in or the value that's being passed in is not a pre-defined value. Also added tests for these.